### PR TITLE
DOCSP-22708 whats new v4.6

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
# Pull Request Info

Updated the What's New section in v4.6 based off of [the engineers release notes](https://github.com/mongodb/node-mongodb-native/releases/tag/v4.6.0).

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-22708>
Staging - <https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-22708-v4.6WhatsNew/whats-new/#what-s-new-in-4.6>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -28,8 +28,31 @@ Learn what's new in:
...
 
-See `v4.6.0 Release Highlights <https://github.com/mongodb/node-mongodb-native/releases/tag/v4.6.0>`__
-on GitHub.
+New features of the 4.6 Node.js driver release include:
+
+TypeScript: ``ChangeStreamDocument``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Ability to define your own custom type for the top level document
+  returned in a ``change`` event.
```
Comment (ccho-mongodb) [inline source/whats-new.txt:37]: Issue: the term is "change event". In the current formatting, it seems like "change" is a type, variable, object, method, collection name, etc. It doesn't appear to be any of these.
Suggestion: remove the formatting for change. If "change" is a type or field, refer to it appropriately and describe what it is.

File: source/whats-new.txt [hunk 2]
```diff
@@ -28,8 +28,31 @@ Learn what's new in:
...
+- Improvements on the type information of the default
```
Comment (ccho-mongodb) [inline source/whats-new.txt:38]: Suggestion: I'm not able to identify more than one improvement listed. Instead, why not just describe the improvement, e.g (I'm not sure this is correct, would need to read the notes).:
"The ChangeStreamDocument now defaults to the option to. union all the possible event types"

File: source/whats-new.txt [hunk 3]
```diff
@@ -28,8 +28,31 @@ Learn what's new in:
...
+  ``ChangeStreamDocument`` to default to union of the possible events.
+- Added the ``to``, ``lsid``, ``txnNumber`` and ``clusterTime`` properties.
```
Comment (ccho-mongodb) [inline source/whats-new.txt:40]: Suggestion: link to the API docs as it may not be obvious what these are.

File: source/whats-new.txt [hunk 4]
```diff
@@ -28,8 +28,31 @@ Learn what's new in:
...
+
+.. note:: 
+
+   If you want to defer the type corrections, use the watch API as
+   follows:
+
+   .. code-block:: typescript
+
+      // 'X' controls the change document type
+      .watch<any, X>()
```
Comment (ccho-mongodb) [inline source/whats-new.txt:50]: Issue: I'm not able to figure out what this note applies to.

File: source/whats-new.txt [hunk 5]
```diff
@@ -28,8 +28,31 @@ Learn what's new in:
...
+
+Server Selection
+~~~~~~~~~~~~~~~~
+
+- Distrubute the server load more evenly across all servers.
```
Comment (ccho-mongodb) [inline source/whats-new.txt:55]: Suggestion: Even distribution of server selection based on load across servers

**Review by ccho-mongodb — CHANGES_REQUESTED** [review]
Good start, some of the items aren't clear to me, so I would like to review the fixes. Thanks!

File: source/whats-new.txt (hunk 4 above)
Comment (kyuan-mongodb) [reply source/whats-new.txt:50]: So the drivers RN mention this as something users need to do if they want to upgrade to this driver version but defer type corrections. It looked/sounded important so I brought it over, but am open to removing it if its not important.

File: source/whats-new.txt [hunk 6]
```diff
@@ -28,8 +28,28 @@ Learn what's new in:
...
+
+TypeScript: ``ChangeStreamDocument``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Ability to define your own custom type for the top level document
+  returned in a change event.
+- The ``ChangeStreamDocument`` type now defaults to union all the
+  possible events.
+- Added `the following properties
+  <{+api+}/modules.html#ChangeStreamDocument>`__ : ``to``,
```
Comment (ccho-mongodb) [inline source/whats-new.txt:41]: Issue: link text doesn't describe the page you're linking to, see the [style guide](https://www.mongodb.com/docs/meta/style-guide/style/links-and-cross-references/)

Suggestion: "Added the following properties to ChangeStreamDocument:" (link ChangeStreamDocument)

File: source/whats-new.txt [hunk 7]
```diff
@@ -28,8 +28,28 @@ Learn what's new in:
...
+TypeScript: ``ChangeStreamDocument``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Ability to define your own custom type for the top level document
+  returned in a change event.
+- The ``ChangeStreamDocument`` type now defaults to union all the
+  possible events.
+- Added `the following properties
+  <{+api+}/modules.html#ChangeStreamDocument>`__ : ``to``, 
+  ``lsid``, ``txnNumber`` and ``clusterTime``.
```
Comment (ccho-mongodb) [inline source/whats-new.txt:42]: Issue: this doesn't seem like something we can expect the reader to understand what the fields are used for without some context.
Suggestion: it appears these are documented in the Server manual, so linking to them would be helpful.

File: source/whats-new.txt [hunk 8]
```diff
@@ -28,8 +28,28 @@ Learn what's new in:
 What's New in 4.6
 -----------------
 
-See `v4.6.0 Release Highlights <https://github.com/mongodb/node-mongodb-native/releases/tag/v4.6.0>`__
-on GitHub.
+New features of the 4.6 Node.js driver release include:
+
+TypeScript: ``ChangeStreamDocument``
```
Comment (ccho-mongodb) [inline source/whats-new.txt:33]: Issue: not sure the technical details are properly communicated.

Suggestion: Given our recent discussion about how we should avoid trying to teach something in the What's New section and go into too much detail, remove this section on ChangeStreamDocument changes and link to the GitHub tag notes. For the link description, explain what [this ticket ](https://jira.mongodb.org/browse/NODE-4059) fixes.

**Review by ccho-mongodb — CHANGES_REQUESTED** [review]

**Pull request merged**